scripts/synapse_port_db

```diff
@@ -505,21 +503,20 @@ class Porter(object):
 
             # Check if all background updates are done, abort if not.
             updates_complete = (
-                yield self.sqlite_store.db.updates.has_completed_background_updates()
+                await self.sqlite_store.db.updates.has_completed_background_updates()
             )
             if not updates_complete:
-                sys.stderr.write(
+                raise Exception(
                     "Pending background updates exist in the SQLite3 database."
                     " Please start Synapse again and wait until every update has finished"
                     " before running this script.\n"
                 )
```

Comment:
Is there a way of quitting with a non-zero exit code that doesn't involve throwing an exception? I'd rather not have a big traceback in the logs, as opposed to nicely printing the rror directly to stderr

Reply:
There wasn't, but I fixed that in https://github.com/matrix-org/synapse/pull/6718/commits/8179108378f73f5ef72ce011da43685aefad31fb